osu.Game/Beatmaps/IBeatSyncProvider.cs

```diff
@@ -14,12 +14,26 @@ namespace osu.Game.Beatmaps
     /// Primarily intended for use with <see cref="BeatSyncedContainer"/>.
     /// </summary>
     [Cached]
-    public interface IBeatSyncProvider
+    public interface IBeatSyncProvider : IHasAmplitudes
     {
+        /// <summary>
+        /// Check whether beat sync is currently available.
+        /// </summary>
+        public bool BeatSyncAvailable => Clock != null;
+
+        /// <summary>
+        /// Whether the beat sync provider is currently in a kiai section. Should make everything more epic.
+        /// </summary>
+        public bool IsKiaiTime => Clock != null && (ControlPoints?.EffectPointAt(Clock.CurrentTime).KiaiMode ?? false);
```

Comment:
Probably want these as `sealed` since they shouldn't have any second implementation:
```suggestion
        /// <summary>
        /// Check whether beat sync is currently available.
        /// </summary>
        sealed bool BeatSyncAvailable => Clock != null;

        /// <summary>
        /// Whether the beat sync provider is currently in a kiai section. Should make everything more epic.
        /// </summary>
        sealed bool IsKiaiTime => Clock != null && (ControlPoints?.EffectPointAt(Clock.CurrentTime).KiaiMode ?? false);
```

Reply:
I'd rather extension methods instead of sealed, but I see no harm in letting them be overrideable.

Reply:
Hmm, shall I just remove these altogether? I thought it would be nice to have them but there's only one usage of each, so I could just move them back inline?

Reply:
I think the usages read better with these methods, so I'm fine with having them. Also, yeah, implementations could override this to change availability of beat sync on certain scenarios, makes sense to leave them unsealed.